sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -278,7 +278,8 @@ case class RegrSlope(left: Expression, right: Expression) extends DeclarativeAgg
     covarPop.mergeExpressions ++ varPop.mergeExpressions
 
   override lazy val evaluateExpression: Expression = {
-    If(covarPop.n === 0.0, Literal.create(null, DoubleType), covarPop.ck / varPop.m2)
+    If(Or(covarPop.n === 0.0, varPop.m2 === 0.0),
```

Comment:
This is a behavior change in ANSI mode but looks reasonable. The two functions are new in 3.4 so no breaking change. @beliefer can you double-check the formular? I'm wondering why we only checked `covarPop.n === 0.0` at the first place.

Reply:
I think we need to remove the check on covarPop - it's fine if that is 0, the slope is just 0. I think it was a typo and was meant to be only varPop.m2 checked for 0

Reply:
Will remove `covarPop`. 

BTW, I think this is a bug fix. Such null behaviour is consistent everywhere in other aggregate functions that use divisions, see also

```scala
scala> spark.conf.set("spark.sql.ansi.enabled", true)

scala> spark.range(1).selectExpr("sum(null)").show()
+---------+
|sum(NULL)|
+---------+
|     null|
+---------+
```

It's more null propagation that failed before. ANSI documentation doesn't mention about null handling from my cursory look so I think we can just keep ti consistent within our codebase

Reply:
At first, I think if the `covarPop.n === 0.0` means `varPop.n === 0.0` and `varPop.m2 === 0.0`.
It seems `varPop.m2 === 0.0` not determined by `varPop.n === 0.0`.